src/export/ValueSetExporter.ts

```diff
@@ -130,6 +135,155 @@ export class ValueSetExporter {
+              containedItem.display = concept.display;
+            }
+            if (concept.designation) {
+              containedItem.designation = cloneDeep(concept.designation);
+            }
+            // remove before adding, since we do not want it to be present multiple times
+            this.removeConcept(containedItem, valueSet.expansion.contains);
+            valueSet.expansion.contains.push(containedItem);
+          });
+          // add known included code systems
+          if (!compose.concept?.length && compose.system) {
+            const codeSystem = referencedCodeSystems[compose.system];
+            const newConcepts = this.getConcepts(codeSystem, codeSystem.concept);
+            // flatten so we can remove each existing concept before adding
+            const newConceptsFlat = flatMap(newConcepts, extractContained);
+            newConceptsFlat.forEach(concept =>
+              this.removeConcept(concept, valueSet.expansion.contains)
+            );
+            valueSet.expansion.contains.push(...newConcepts);
+          }
+        });
+        valueSet.compose.exclude?.forEach(compose => {
+          // remove specific excluded codes
+          compose.concept?.forEach(concept => {
+            this.removeConcept(
+              { system: compose.system, code: concept.code },
+              valueSet.expansion.contains
+            );
+          });
+          // remove known excluded code systems
+          // this is a little unusual to do, but we'll allow it.
+          if (!compose.concept?.length && compose.system) {
+            const codeSystem = referencedCodeSystems[compose.system];
+            const conceptsToRemove = flatMap(this.getConcepts(codeSystem, codeSystem.concept));
+            conceptsToRemove.forEach(concept =>
+              this.removeConcept(concept, valueSet.expansion.contains)
+            );
+          }
+        });
+      }
+    }
+  }
+
+  private getConcepts(system: any, concepts: any[]): ValueSetExpansionContains[] {
+    return concepts.map(concept => {
+      const contained: ValueSetExpansionContains = {
+        system: system.url,
+        code: concept.code
+      };
+      if (system.version) {
+        contained.version = system.version;
+      }
+      if (concept.display) {
+        contained.display = concept.display;
+      }
+      if (concept.concept?.length > 0) {
+        contained.contains = this.getConcepts(system, concept.concept);
+      }
+      return contained;
+    });
+  }
+
+  private removeConcept(target: ValueSetExpansionContains, concepts: ValueSetExpansionContains[]) {
+    const targetIndex = concepts.findIndex(concept => {
+      return concept.code === target.code && concept.system === target.system;
+    });
+    if (targetIndex > -1) {
+      const [removedConcept] = concepts.splice(targetIndex, 1);
+      if (removedConcept.contains?.length) {
+        concepts.splice(targetIndex, 0, ...removedConcept.contains);
+      }
```

Comment:
So if you remove a top-level concept that has child concepts, the children all get bumped up to the same level the removed code used to be at?  Am I reading that right?

Reply:
That's correct. This is the behavior I observed when testing with the IG publisher.